evm/src/cpu/columns/general.rs

```diff
@@ -91,5 +113,32 @@ pub(crate) struct CpuLogicView<T: Copy> {
     pub(crate) output: [T; 16],
 }
 
+#[derive(Copy, Clone)]
+pub(crate) struct CpuJumpsView<T: Copy> {
+    // Assuming a limb size of 32 bits.
+    pub(crate) input0: [T; 8],
+    pub(crate) input1: [T; 8],
+    pub(crate) output: [T; 8],
+
+    pub(crate) input0_upper_sum_inv: T,
+    pub(crate) input0_upper_zero: T,
+
+    pub(crate) dst_valid: T, // TODO: populate this (check for JUMPDEST)
+    pub(crate) dst_valid_or_kernel: T,
+    pub(crate) input0_jumpable: T,
```

Comment:
I'm a bit confused about the meaning of this. The logic seems to be `input0_jumpable = dst_valid_or_kernel AND input0 = 0`?

Reply:
It’s `input0.bits[32..256] = 0`. If the high bits are nonzero, then it’s not a valid jump address (because there can’t be any code that far into memory), and it also won’t fit in our program counter.